plugins/woocommerce-blocks/assets/js/blocks/product-collection/utils.tsx

```diff
@@ -154,18 +154,71 @@ export const addProductCollectionToQueryPaginationParentOrAncestor = () => {
 	}
 };
 
+/**
+ * Get the preview message for the Product Collection block based on the usesReference.
+ * There are two scenarios:
+ * 1. When usesReference is product, the preview message will be:
+ * 	  "Actual products will vary depending on the product being viewed."
+ * 2. For all other usesReference, the preview message will be:
+ *    "Actual products will vary depending on the page being viewed."
+ *
+ * This message will be shown when the usesReference isn't available on the Editor side, but is available on the Frontend.
+ */
+export const getUsesReferencePreviewMessage = (
+	usesReference: string[],
+	location: WooCommerceBlockLocation
+) => {
+	if ( ! ( Array.isArray( usesReference ) && usesReference.length > 0 ) ) {
+		return '';
+	}
+
+	if ( usesReference.includes( location.type ) ) {
+		/**
+		 * Block shouldn't be in preview mode when:
+		 * 1. Current location is archive and termId is available.
+		 * 2. Current location is product and productId is available.
+		 *
+		 * Because in these cases, we have required context on the editor side.
+		 */
+		if (
+			( location.type === LocationType.Archive &&
+				location.sourceData?.termId ) ||
+			( location.type === LocationType.Product &&
+				location.sourceData?.productId )
```

Comment:
I don't know if it's possible for `termId` or `productId` to be `0` but if so, we should check if it has value

```suggestion
			( location.type === LocationType.Archive &&
				location.sourceData?.termId != null ) ||
			( location.type === LocationType.Product &&
				location.sourceData?.productId != null )
```

Reply:
I see you left a thumbup but I don't see it changed 😅 if you agree, please add this check!

Reply:
I have added the check in [this commit](https://github.com/woocommerce/woocommerce/pull/49796/commits/ae96a19fbbd4c43a9279c8cca21a93445f80e347). Isn't that what you were looking for? Maybe I am missing something. 😅

Reply:
I missed "Outdated" label above the code snippet... Need to focus more 🙈 

<img width="307" alt="image" src="https://github.com/user-attachments/assets/a38a1af2-61a9-4ccf-856d-c3209d4913f7">

All good!